cmd/antrea-agent/agent.go

```diff
@@ -148,12 +149,16 @@ func run(o *Options) error {
 		klog.InfoS("enableIPSecTunnel is deprecated, use trafficEncryptionMode instead.")
 		encryptionMode = config.TrafficEncryptionModeIPSec
 	}
+	_, ipsecAuthenticationMode := config.GetIPsecAuthenticationModeFromStr(o.config.IPsec.AuthenticationMode)
```

Comment:
We should validate AuthenticationMode is a valid value in options.go

Reply:
added one to Options.validate(). thanks!